### PR TITLE
Allow alternative parameters for dexsearches

### PR DESCRIPTION
**Pull request by tmagicturtle**
Maybe I'm not understanding something or there's some non-obvious reason for it, but with the new pivot search, I didn't really understand why it had special coding to allow only "batonpass" (and not "baton pass" or other search options). I especially didn't understand why it didn't just use the normal "or" search (e.g. flash fire|baton pass). This got me wondering about the other searches disallowing alternatives. Surely there's a use for a search like "recovery|regenerator" and I don't really see why it specifically disallowed alternatives.

The only other thing I could not figure out was allowing alternatives for mega, fully evolved, and gmax. Surely there's a use for "mega|gmax" or "fully evolved|def>100" or similar. Again, unless there's a reason I'm not aware of, I think these are artificial limitations on the dexsearch function and should be removed.

### CODE REVIEW

File: server/chat-plugins/datasearch.ts [hunk 1]
```diff
@@ -806,17 +800,9 @@ function runDexsearch(target: string, cmd: string, canAll: boolean, message: str
...
-				if (parameters.length > 1) {
-					if (parameters[1].trim().toLowerCase() === 'batonpass') {
-						includeBatonPass = true;
-					} else {
-						return {error: "The parameter 'pivot' cannot have alternative parameters other than 'batonpass'."};
-					}
-				}
 				for (const move in Dex.data.Moves) {
 					const moveData = Dex.getMove(move);
-					if (moveData.selfSwitch && (includeBatonPass || moveData.id !== 'batonpass')) {
```
Comment (KrisXV) [inline server/chat-plugins/datasearch.ts:819]: why are you removing the baton pass parameter?

Comment (tmagicturtle) [reply server/chat-plugins/datasearch.ts:819]: That's the way the pivot parameter already works. If we're going to include baton pass that's fine, but the decision to not include it wasn't mine.

Comment (tmagicturtle) [reply server/chat-plugins/datasearch.ts:819]: The reasoning at https://www.smogon.com/forums/threads/make-pivot-moves-a-ms-category.3672772/ is that most formats ban Baton Pass, so if BP was included, searching by pivot would have little value.

**Comment by tmagicturtle** [comment]
Just checking if there's any additional thoughts on this, it's been some time.

**Comment by tmagicturtle** [comment]
I am asking once again for your review and support.

**Review by AnnikaCodes — APPROVED** [review]
This seems fine to me.

**Comment by AnnikaCodes** [comment]
You have to fix the merge conflicts, though.

File: server/chat-plugins/datasearch.ts [hunk 2]
```diff
@@ -836,17 +830,9 @@ function runDexsearch(target: string, cmd: string, canAll: boolean, message: str
...
-					if (parameters[1].trim().toLowerCase() === 'batonpass') {
-						includeBatonPass = true;
-					} else {
-						return {error: "The parameter 'pivot' cannot have alternative parameters other than 'batonpass'."};
-					}
-				}
 				for (const move in mod.data.Moves) {
-					const moveData = mod.moves.get(move);
-					if (moveData.selfSwitch && (includeBatonPass || moveData.id !== 'batonpass')) {
+					const moveData = mod.getMove(move);
```
Comment (Distrib-ps) [inline server/chat-plugins/datasearch.ts:834]:
```suggestion
					const moveData = mod.moves.get(move);
```

**Comment by AnnikaCodes** [comment]
Sorry I missed this, it looks good. Thanks!

**Pull request merged**